newspaper/urls.py

```diff
@@ -245,8 +245,13 @@ def url_to_filetype(abs_url):
         path = path[:-1]
     path_chunks = [x for x in path.split('/') if len(x) > 0]
     last_chunk = path_chunks[-1].split('.')  # last chunk == file usually
-    file_type = last_chunk[-1] if len(last_chunk) >= 2 else None
-    return file_type or None
+    if len(last_chunk) < 2:
+        return None
+    file_type = last_chunk[-1]
+    # Assume that file extension is maximum 5 characters long
+    if len(file_type) <= 5 or file_type.lower() in ALLOWED_TYPES:
```

Comment:
@codelucas `5` is a quick-and-dirty heuristic, and it should be kept local. Moving it to constants would only give it an undeserved air of authority.

The proper solution is to use a extension blacklist and HEAD requests for uncertain cases.